### ISSUE TITLE
feature:  ability to swap article in an order with an alternative

### ISSUE THREAD

**Issue by carchrae** [foodcoops/foodsoft#646]
i have already implemented a version of this, so take this as my intention to submit a PR soon.

the price and availability of fresh produce changes often.  our co-op has a dilemma here; we either allow people to order over a very short time (typically 2 days due to our delivery date) or we put the order up and then update the prices/availability of the articles while the order is still open.

updating prices is not really a problem and happens via the `synchronise` button.

one problem when updating prices is what happens when an item is no longer available.  this happens.  the current behaviour will not allow you to update prices until you delete it from the current order.  this is extremely tedious.  instead, i have modified the sync code to change the title of the item, appending "UNAVAILABLE" to the name.  this may seem crude but this works surprisingly well.  the ordering person can go through and change the item to something which is available, and in the case that they forget to do it - both the co-op members and the supplier can see that the item is not available.  

however, it would be useful to

1) let members know about the price/supplier changes (a summary of changes to their order)
2) make it easy for the ordering person to choose an alternative 
3) indicate when a cheaper alternative is available (based on item name + origin)
4) suggest that another item's case size better matches the co-op demand (eg, 24 CELERY is better than 36 based on the order)
5) be able to translate units ordered (eg, 1 x 5LB bag into 5 x 1LB bags) - this is manual and all checks for equivalency are manual.
6) getting very ambitious here, but have the co-op site be able to automatically choose the best case to match the co-op demand.  (this is dreaming i think)

i attach a screen shot of my current implementation of the "swap" page.  

![image](https://user-images.githubusercontent.com/959825/55346441-566f2500-5467-11e9-82e1-597d5e32a8a0.png)

**Comment by carchrae** [comment]
i have discovered one problem with my approach in changing the name.  old orders that contain articles that are no longer available will start to have "unavailable" tagged on them.  i don't think this is a big problem, but it perhaps is a good reason to use the "available" field instead and then update the UI in all places to ensure that this is clearly marked

**Comment by wvengen** [comment]
Yes, this is a feature that I've heard people wishing as well. For you it happens during the order cycle, I've also seen it happen after the order was closed.

If it happens during the order, the order screen makes sense to work with. If there are not so many articles, it could be a button in the order's article overview (e.g. with a modal). If there are many, then perhaps it could be a batch action there. Then the screen that you gave an example of.
Some user notification here would be useful, perhaps to opt-in/opt-out of the change.

If articles are discontinued after the order is closed, the receive screen seems a logical place. There you'd indicate that a certain article didn't come in, and would add a replacement article (I've seen suppliers giving something else instead when something wasn't available, or sometimes it might be agreed upon when picking up the vegetables from the farmer).

**Comment by carchrae** [comment]
yes, we mostly use this while it is open.  but it does happen when it is closed too - as the supplier offers alternatives.  most of the time, it is just the farm that changes, but sometimes the case size changes too.  as long as the units are the same, this has been working well for us.  the case size changing is actually great, because sometimes you can find a better sized case to fit the demand and then it re-distributes just as the received page does. (this is related to our discussion on variants and partial cases).  

on our last order, the person ordering swapped 10 x 4lb bags for 100 x item.  i'm not sure how much of that conversion swapping should support - it isn't always easy/possible to translate, eg, 10x5lb bags into 25x2lb bags may end up with fractions of a bag.  (again, if people ordered by lb then this would be ok if there was tolerance)

when the order is closed, we usually just update the price on the balancing page.  

on the discontinued aspect - this happens while our orders are open.  our produce supplier changes the prices every 3-4 days.  to make it easier for our members to order, we post the order a week before, and then update the prices.  before we did this it was always a mad dash in a short time to get the order up and people to order.
